plugin/azure/azure.go

```diff
@@ -196,6 +225,87 @@ func updateZoneFromResourceSet(recordSet azuredns.RecordSetListResultPage, zName
 	return newZ
 }
 
+func updateZoneFromPrivateResourceSet(recordSet privateAzureDNS.RecordSetListResultPage, zName string) *file.Zone {
+	newZ := file.NewZone(zName, "")
+
+	for _, result := range *(recordSet.Response().Value) {
+		resultFqdn := *(result.RecordSetProperties.Fqdn)
+		resultTTL := uint32(*(result.RecordSetProperties.TTL))
+		if result.RecordSetProperties.ARecords != nil {
```

Comment:
why is all this code repeated? Please factor it out a bit and reuse it for both private and public

Reply:
these are distinct types in the auzre sdk codebase. 
Without generics, I am not sure how to refactor this, any ideas?

Reply:
I don't see it, it looks idential to the public ones  `*(recordSet.Response().Value) ` how it this different?

Reply:
the type of `privatedns.RecordSetListResultPage` and the otherone is `publicdns.RecordSetListResultPage``